src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,880 @@
+        try:
+            kwargs = children["pressure"]
+        except (KeyError, TypeError):
+            logging.info("Orsay was not given a 'pressure' child")
+        else:
+            self._pressure = vacuumChamber(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._pressure)
+
+        # create the pumping system child
+        try:
+            kwargs = children["pumping-system"]
+        except (KeyError, TypeError):
+            logging.info("Orsay was not given a 'pumping-system' child")
+        else:
+            self._pumpingSystem = pumpingSystem(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._pumpingSystem)
+
+        # create the UPS child
+        try:
+            kwargs = children["ups"]
+        except (KeyError, TypeError):
+            logging.info("Orsay was not given a 'ups' child")
+        else:
+            self._ups = UPS(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._ups)
+
+    def on_connect(self):
+        """
+        Defines direct pointers to server components and connects parameter callbacks for the Orsay server.
+        Needs to be called after connection and reconnection to the server.
+        """
+        self.datamodel = self._device.datamodel
+        self.datamodel.HybridPlatform.ProcessInfo.Subscribe(self._updateProcessInfo)
+
+        self.update_VAs()
+
+    def update_VAs(self):
+        """
+        Update the VA's. Should be called after reconnection to the server
+        """
+        self._updateProcessInfo()
+
+    def _connection_monitor(self):
+        """
+        Once in a while, check the connection to the Orsay server, reconnect if needed and update all VA's
+        """
+        while not self._stop_connection_monitor.is_set():
+            if self._device.HttpConnection._HTTPConnection__response is None or \
+                    self._device.MessageConnection.Connection._HTTPConnection__response is None:
+                self.state._set_value(HwError("Connection to Orsay server lost. Trying to reconnect..."),
+                                      force_write=True)
+                self._device.HttpConnection.close()  # close the current connection
+                self._device.MessageConnection.Connection.close()
+                self._device = None  # destroy the current connection object
+                try:
+                    self._device = Connection(self._host)
+                    time.sleep(1)
+                    self.on_connect()
+                    for child in self.children.value:
+                        child.on_connect()
+                    self.state._set_value(model.ST_RUNNING, force_write=True)
+                except Exception as ex:
+                    logging.exception("Trying to reconnect to Orsay server: %s." % ex)
+            else:
+                self.update_VAs()
+                for child in self.children.value:
+                    child.update_VAs()
```

Comment:
I just realised this will actually never fail due to connection errors. The update_VAs methods read parameters from the ConsoleClient component, which will stay in existence until the orsayComponent detects that there is no connection. Additionally the update_VAs will never be called when the ConsoleClient component does not exist. So if the connection would (forcefully) disappear during a call to an update_VAs method, the update would succeed just fine, but potentially with outdated information.
I added a try block anyway to catch other exceptions, might they occur.